File: src/1_endianness/Endianness.java
import java.nio.ByteOrder;

public class Endianness
{
    public static void main(String argv[])
    {
        if (ByteOrder.nativeOrder().equals(ByteOrder.BIG_ENDIAN))
            System.out.println("Big Endian");
        else
            System.out.println("Little Endian");
    }
}

File: src/1_endianness/endianness.c
#include <stdio.h>

int main(int argc, char**argv)
{
	int i=1;
	char* p = (char*) &i;

	if((int) *p==i)
		printf("Little Endian \n");
	else
		printf("Big Endian \n");
}
